### PR TITLE
Implement np.linalg.inv using a QR decomposition.

### LINKED ISSUE #44: np.linalg.inv support
I've been trying to implement Gaussian Processes Regression, which require the calculation of a matrix inverse. With regular numpy I would use `np.linalg.inv`, but I can't find this function back in jax.

Everything else is working as expected, and I can use `np.linalg.inv` for basic calculations.
Unfortunately, the use of `np.linalg.inv` keeps me from using `grad` to calculate gradients, which would be the most exciting part of the whole implementation!

I would love to contribute a PR if someone can tell me where to start.

### PR DESCRIPTION
**Pull request by hawkinsp**
An LU decomposition would probably be preferable; we can switch the implementation when we have an LU decomposition.

Fixes #44.

### CODE REVIEW

**Review by mattjj — APPROVED** [review]
Love it!
